Individual page/profile ui update

dannysperry opened the pull request:
[Trello Card](https://trello.com/c/dzU71Z20/2114-individual-view-ui)

**Description**
Follow the design from the invisionapp for the individual profile view.

This covers UI redesign for Individuals and redux flow for the addition of goals and needs

cvkumar commented on src/redux/actions/index.js:
```diff
@@ -300,3 +296,68 @@ export function getHomelessTimeline(token, beaconId) {
     }
 }
 
+function fetchHomelessGoals() {
```
Nice work on these

dannysperry replied:
🙇

cvkumar commented on src/components/individuals/profile/status/index.js:
```diff
@@ -0,0 +1,150 @@
+        const { classes, homelessInfo, homelessGoals, homelessNeeds } = this.props
+        
+        let title = null
+        if ((homelessGoals && homelessGoals.length > 0) || (homelessNeeds && homelessNeeds.length > 0)) {
```
I have this little helper if you want to use it: https://github.com/givesafe/lifecare-webapp/blob/ac306e6e8a674800883570a60f73b3bf5c6e8594/src/util/index.js#L1

dannysperry replied:
👍

cvkumar commented on src/redux/constants/constants.js:
```diff
@@ -26,3 +32,6 @@ export const UNAUTHORIZED_USER = 'UNAUTHORIZED_USER'
 export const RESET_API_ERROR = 'RESET_API_ERROR'
+
+// export const SAMARITAN_URL = 'https://givesafe-staging.herokuapp.com'
+export const SAMARITAN_URL = 'http://localhost:3001'
```
With the PR I just merged, you can now use `.env`! Just set .env variable: `REACT_APP_SAMARITAN_URL` in it. And then it'll get read from here: https://github.com/givesafe/lifecare-webapp/blob/9615bf1878b885c0dd2cae96742e296a3c24afc9/src/settings.js#L1

cvkumar replied:
Let's remove the hardcoded URL if okay.

dannysperry replied:
I noticed your updates and I removed SAMARITAN_URL from redux constants.

dannysperry commented:
I made a couple updates since your review. 

I updated the goals and needs fetching to be dispatched with homeless info, we may want to do the same thing for timeline. I pulled your updates in and fixed usage of SAMARITAN_URL.

I'm going to do some tests with prod like you suggested and work on px vs %/em.

Highwind360 commented on src/components/individuals/profile/lifecare.js:
```diff
@@ -0,0 +1,107 @@
+
+}
+
+function calculateDaysSinceLastCheckin(lastCheckin) {
```
Perhpas `getDaysLeftEnabled` or `getDaysUntilNextCheckinRequired` would be a more correct name for this.

dannysperry replied:
renamed and fixed the function so that it returns 0 if overdue

Highwind360 commented on src/components/individuals/profile/lifecare.js:
```diff
@@ -0,0 +1,107 @@
+    const msSinceLastCheckin = Date.now() - Date.parse(lastCheckin)
+    const daysSinceLastCheckin = msSinceLastCheckin / msPerDay
+
+    return parseInt(30 - daysSinceLastCheckin)
```
Is there a good place to put constants in RN? `30` seems like information we may want to use in more than one place, or at least would want to have in a configuration file, somewhere.

dannysperry replied:
I added a constant here as suggested. Right now we are using `src/settings`

dannysperry commented on src/components/individuals/index.js:
```diff
@@ -99,41 +98,12 @@ class Individuals extends React.Component {
+    onSelect(beaconId) {
         let auth = AuthManager.getInstance()
-        // TODO: The beacon holder beacon id must be passed here
+        // TODO: The beacon holder beacon beaconId must be passed here
```
correct 👍

dannysperry commented on src/components/individuals/index.js:
```diff
@@ -197,22 +156,26 @@ class Individuals extends React.Component {
+            //             </div>
+            //         </Card>
+            //     </div>
+            // )
```
This i'm not sure about. This whole individual section previous had a more "data dashboard" set of functionality. We may add these finer details back in eventually but for now we are getting some basic partner functionality setup for Next City MVP

dannysperry commented:
@Highwind360 Revisions requested applied to a few more commits

Highwind360 commented on src/components/individuals/profile/lifecare.js:
```diff
@@ -86,13 +88,14 @@ class Lifecare extends Component {
 
-    return parseInt(30 - daysSinceLastCheckin)
+    if (daysSinceLastCheckin > CHECKIN_CYCLE_DAYS) return 0
+    return daysSinceLastCheckin
```
Just confirming that the intent was to change the method's behavior from returning the days until the next checkin is required, to instead returning the days since the last checkin, which for some reason becomes 0 after they're due for another?
